migrations/frontend/1653596521/up.sql

```diff
@@ -0,0 +1,7 @@
+ALTER TABLE changesets
+    ADD COLUMN IF NOT EXISTS detached_at timestamp with time zone;
+
+CREATE INDEX IF NOT EXISTS
+    changesets_detached_at
+    ON
+        changesets (detached_at);
```

Comment:
```suggestion
        changesets (detached_at) WHERE detached_at IS NOT NULL;
```

Would that help limit the index size a bit further?

Reply:
I'm not sure about the index size. But if this were to be non-null, what value should we set this as and set all existing rows to? `0001-01-01` (zero)?

Reply:
No I meant isn't a partial index (like above) a little smaller since it doesn't have to store a huge number of `NULL` values? (not a DB expert but it sounds to me like that could be the case). Not saying we should change to non nullable, just adding the partial constraint on the index :)